internal/server/logbuffer/merge.go

```diff
@@ -0,0 +1,146 @@
+package logbuffer
+
+import (
+	"io"
+	"time"
+)
+
+// Merger can combine multiple log streams into one stream. It presumes
+// each stream emits entries in time order, and then weaves the entries
+// together to create a total time ordered stream.
+type Merger struct {
+	readers []MergeReader
+	heads   []TimedEntry
+}
+
+// TimedEntry is the interface each input returns entries in
+type TimedEntry interface {
+	// Time is the associated time of the value. This is used to sort
+	// the entry against other entries from other inputs.
+	Time() time.Time
+
+	// Value returns the log entry itself. This allows time addition
+	// wrappers to return their original value.
+	Value() interface{}
```

Comment:
nit: `Value() Entry` may be clearer here (Even though it's also an empty interface{})

Reply:
The reason it's not Entry is if it were, then the implementors would have to import logbuffer to satisfy the interface.

Reply:
Oh, I missed that, good point!